src/topp/PercolatorAdapter.cpp

```diff
@@ -0,0 +1,1026 @@
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Mathias Walzer $
+// $Authors: Andreas Simon, Mathias Walzer, Matthew The $
+// --------------------------------------------------------------------------
+#include <OpenMS/config.h>
+
+#include <OpenMS/FORMAT/FileHandler.h>
+#include <OpenMS/FORMAT/FileTypes.h>
+#include <OpenMS/DATASTRUCTURES/StringListUtils.h>
+#include <OpenMS/APPLICATIONS/TOPPBase.h>
+#include <OpenMS/FORMAT/MzIdentMLFile.h>
+#include <OpenMS/FORMAT/IdXMLFile.h>
+#include <OpenMS/FORMAT/CsvFile.h>
+#include <OpenMS/CONCEPT/ProgressLogger.h>
+#include <OpenMS/CONCEPT/Constants.h>
+#include <OpenMS/ANALYSIS/ID/TopPerc.h>
+#include <QtCore/QFile>
+#include <QtCore/QDir>
+#include <QtCore/QProcess>
+
+#include <iostream>
+#include <cmath>
+#include <string>
+#include <set>
+
+#include <boost/algorithm/clamp.hpp>
+#include <typeinfo>
+
+using namespace OpenMS;
+using namespace std;
+
+//-------------------------------------------------------------
+//Doxygen docu
+//-------------------------------------------------------------
+
+/**
+  @page TOPP_PercolatorAdapter PercolatorAdapter
+
+  @brief PercolatorAdapter facilitates the input to, the call of and output integration of Percolator.
+  Percolator (http://per-colator.com/) is a tool to apply semi-supervised learning for peptide
+  identification from shotgun proteomics datasets.
+
+  @experimental This tool is work in progress and usage and input requirements might change.
+
+  <center>
+    <table>
+        <tr>
+            <td ALIGN = "center" BGCOLOR="#EBEBEB"> pot. predecessor tools </td>
+            <td VALIGN="middle" ROWSPAN=2> \f$ \longrightarrow \f$ PercolatorAdapter \f$ \longrightarrow \f$</td>
+            <td ALIGN = "center" BGCOLOR="#EBEBEB"> pot. successor tools </td>
+        </tr>
+        <tr>
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=1> @ref UTILS_PSMFeatureExtractor </td>
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=1> @ref TOPP_IDFilter </td>
+        </tr>
+    </table>
+  </center>
+  <p>Percolator is search engine sensitive, i.e. it's input features vary,
+depending on the search engine. Must be prepared beforehand. If you do not want
+to use the specific features, use the generic-feature-set flag. Will incorporate
+the score attribute of a PSM, so be sure, the score you want is set as main
+score with @ref TOPP_IDScoreSwitcher . Be aware, that you might very well
```

Comment:
Does not get transformed to a link if 'TOPP_IDScoreSwitcher.' - at least with my doxygen